qcodes/instrument_drivers/agilent/E8527D.py

```diff
@@ -10,26 +10,75 @@
 class Agilent_E8527D(VisaInstrument):
     """
     This is the QCoDeS driver for the Agilent_E8527D signal generator.
-
     This driver will most likely work for multiple Agilent sources.
-
     This driver does not contain all commands available for the E8527D but
     only the ones most commonly used.
     """
     def __init__(self, name: str, address: str,
-                 step_attenuator: bool = False,
```

Comment:
I am not sure if we can just remove this step_attenuator since that may break some other users use of the code

Reply:
ok. as you wish. But it will not be used within the driver, it will only be an argument of the init function.

Reply:
I suggest that you do the following to avoid breaking users code.

keep step_attenuator and add something like the following:

```
    def __init__(self, name: str, address: str,
                 step_attenuator: Optional[bool] = None,
                 terminator: str = '\n',
                 **kwargs: Any) -> None:

    if step_attenuator is not None:
         warnings.warn("step_attenuator argument to E8527D is deprecated and has no effect. It will be removed in the future.)
```